### ISSUE TITLE
Carthage build reachability framework is outdated

### ISSUE THREAD

**Issue by graylikeme** [awslabs/aws-mobile-appsync-sdk-ios#76]
**Describe the bug**
When building with Carthage the version of Reachability that is being downloaded is 3, but the sdk binary attempts to use functions from v4

`dyld: Symbol not found: _$S12ReachabilityAAC24allowsCellularConnectionSbvpWvd
  Referenced from: /private/var/containers/Bundle/Application/54737AA0-D181-4133-830C-E863D0AA335B/Pear-DEV.app/Frameworks/AWSAppSync.framework/AWSAppSync
  Expected in: /private/var/containers/Bundle/Application/54737AA0-D181-4133-830C-E863D0AA335B/Pear-DEV.app/Frameworks/Reachability.framework/Reachability
 in /private/var/containers/Bundle/Application/54737AA0-D181-4133-830C-E863D0AA335B/Pear-DEV.app/Frameworks/AWSAppSync.framework/AWSAppSync
`

**Environment(please complete the following information):**
 - AppSync SDK Version: [2.6.22]
 - Dependency Manager: [Carthage]
 - Swift Version : [4.2]

**Additional context**
The problem seems to be the version specified in Cartfile (specified to be compatible with v 0.4.0). Needs an update to the latest

**Comment by pawelurb** [comment]
For people currently struggling with this issue, there's a workaround for this problem that you can apply by explicitly defining version of Reachability in your Cartfile:

```
github "awslabs/aws-mobile-appsync-sdk-ios" ~> 2.6.0
github "ashleymills/Reachability.swift" "v4.2.1"
```

Hint: if you're building with Swift 4.2 then [v4.3.0 has been released](https://github.com/ashleymills/Reachability.swift/releases), so you can update your Cartfile accordingly.

However, I also agree that this issue should be addressed in the library source files.

**Comment by kiroskirin** [comment]
@pawelurb Still got error even doing as your workaround.

**Comment by pawelurb** [comment]
@kiroskirin can you paste the content of your Cartfile please and let me know which version of Swift are you using?

**Comment by peterfennema** [comment]
Same issue here, swift 4.2. The workaround of @pawelurb does not work. 
The v3 version of Reachability is downloaded, the 4.3.0. version is ignored.

Cartfile:

```
github "awslabs/aws-mobile-appsync-sdk-ios" == 2.6.22
github "ashleymills/Reachability.swift" == 4.3.0

```

**Comment by pawelurb** [comment]
@kiroskirin @peterfennema I'll try to investigate this issue, set up a sample repository and get back to you in a couple of hours.

**Comment by kiroskirin** [comment]
@pawelurb At first here my Carfile

```
github "awslabs/aws-mobile-appsync-sdk-ios"
github "ashleymills/Reachability.swift"
```

But after I test your workaround again

```
github "awslabs/aws-mobile-appsync-sdk-ios"
github "ashleymills/Reachability.swift" "v4.2.1"
```

It seem to work 👍

**Comment by kiroskirin** [comment]
@pawelurb Now I'm struggling with this issue #83 Can you look into that.

Thank you

**Comment by graylikeme** [comment]
@kiroskirin I also had this problem, I ended up giving up fighting with their carthage build and just used frameworks I got from cocoapods as drop-ins.

After you will solve the SQLite problem, you'll get a crash elsewhere.

**Comment by kiroskirin** [comment]
@graylikeme Okay, so I have no choice then 😞

**Comment by palpatim** [comment]
Hi all,

Version 2.9.1 of the SDK includes Pods in the source code, allowing devs who consume from source (e.g., directly including source, or using `carthage update --no-use-binaries` to run without CocoaPod installation. That should make it easier to build from source and work around the Swift incompatibility issue.

However, there may be issues with framework incompatibility if the host app is using different versions of Reachability and SQLite. We recently updated those versions:

```
github "stephencelis/SQLite.swift" ~> 0.11.5
github "ashleymills/Reachability.swift" ~> 4.3.0
```

Which should be the latest versions of each library.

Please note that these are really just advisory; even though the Cartfile specifies versions (and would potentially download and build source during a `carthage update`), the AppSync project will be built from source using the included dependencies in the `Pods` directory.

One solution we are considering to mitigate the confusion is to remove the dependencies from the Cartfile altogether, since it doesn't really provide any build-time value, and the documentation value could be handled with a comment-only Cartfile referencing the project's `Pods` directory and `Podfile.lock`. The resulting Cartfile would look something like:

```
# Dependencies for this project are managed via CocoaPods and checked into source.
# Please see Podfile.lock and Pods directory for the in-use versions.
```

However, there may be tooling I'm unaware of that relies on an up-to-date Cartfile to do dependency version validation, so we haven't settled on that as *the* solution yet. I'd welcome any feedback on this proposal.

**Comment by palpatim** [comment]
I've replaced the actual Cartfile contents with a comment as described above. Carthage still builds the project locally (and as a side benefit, no longer downloads the specified dependencies that would be extraneous in any case). See PR #158

And this should remove confusion about which dependency versions are actually in use. Combined with including Podfiles in the source, this should greatly reduce the problems building AppSync with Carthage.

**Comment by palpatim** [comment]
I've withdrawn #158 since it would break transitive dependencies. Because they need to be kept in sync, I'm also not adding the comment that the Podfile is the source of truth--it's up to us to keep them in sync.

Since we fixed the base issue on PR #151, released with 2.9.1, and  have no further enhancements to clarify the Cartfile intent, I'm resolving this issue.
